provider/datagen/src/error.rs

```diff
@@ -2,125 +2,23 @@
 // called LICENSE at the top level of the ICU4X source tree
 // (online at: https://github.com/unicode-org/icu4x/blob/main/LICENSE ).
 
-use displaydoc::Display;
-use icu_locid::LanguageIdentifier;
 use icu_provider::{DataError, DataErrorKind};
-use std::path::{Path, PathBuf};
-
-#[non_exhaustive]
-#[derive(Display, Debug)]
-pub enum DatagenError {
```

Comment:
Praise: Although I had originally anticipated moving all of `icu_datagen` to `eyre`, I think it makes sense today to move it all to `DataError` instead, since so much work happens inside the DataProvider impls, which return `DataError`.